runbotics/runbotics-orchestrator-ui/src/src-app/credentials/getRequiredCredentialsTypesInProcess.ts

```diff
@@ -0,0 +1,18 @@
+import { BpmnElementType } from 'runbotics-common';
+import BpmnModelerType from 'bpmn-js/lib/Modeler';
+import { BPMNElementRegistry } from '#src-app/views/process/ProcessBuildView/Modeler/helpers/elementParameters';
+import actions from '#src-app/Actions';
+import { ActionCredentialType } from '#src-app/credentials/actionCredentialType.enum';
+
+export const getRequiredCredentialsTypesInProcess = (modeler: BpmnModelerType): ActionCredentialType[] => {
+    const elements = modeler.get('elementRegistry')._elements as BPMNElementRegistry;
+
+    const credentialTypesWithDuplicates = Object.values(elements)
+        .filter(element => element.element.type === BpmnElementType.SERVICE_TASK)
+        .map(serviceTask => {
+            return actions[serviceTask.element.businessObject.actionId].credentialType;
+        })
+        .filter(credentialType => !!credentialType);
```

Comment:
you can make one reduce that returns acc if !actions[serviceTask.element.businessObject.actionId].credentialType
and [...acc, actions[serviceTask.element.businessObject.actionId].credentialType] otherwise